benchmarks/mm_prod.cpp

```diff
@@ -0,0 +1,52 @@
+#include <boost/numeric/ublas/matrix.hpp>
+#include <boost/program_options.hpp>
+#include "prod.hpp"
+#include <complex>
+#include <string>
+
+namespace po = boost::program_options;
+
+template <typename T>
+void benchmark(std::string const &type)
+{
+  using matrix = ublas::matrix<T, ublas::basic_row_major<>>;
+  prod<matrix(matrix, matrix)> p("prod(matrix<" + type + ">)");
+  p.run(std::vector<long>({1, 2, 4, 8, 16, 32, 64, 128, 256, 512}));//, 1024}));
+}
+
+int main(int argc, char **argv)
```

Comment:
I am not sure if the benchmarks should be runtime-variable in terms of the data type and not in terms the vector, matrix and tensor size.

We might need a more generic approach where we can select from one main functions to test and maybe including the type.

Reply:
I'm not sure I understand what you have in mind. Can you elaborate ?

Reply:
Let me come back to this (sorry for the high latency):
I disagree about the distinction between "contributors" and "end-users". I expect benchmarks to be of most value during development, where someone may want to run (and compare !) a specific set of benchmarks.
And to support that, I need benchmarks to be runnable individually, not en bloc.
(In fact, I'd argue the same way for tests: it's good to have a way to run an entire test suite with a single command, but ultimately it must be possible to invoke individual tests, too, especially if they fail and you want to investigate why.)